### PR TITLE
[vtctldserver] Migrate `Backup` RPC to vtctldserver/client

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This PR migrates the `Backup` RPC to vtctldserver, which **makes this our first streaming RPC**, building on top of my work in #9116. 

### Testing

<details><summary>Successful Backup</summary>

```
➜  local git:(andrew/vtctldserver-backup) ✗ vtctldclient --server ":15999" Backup zone1-101
commerce/0 (zone1-0000000101): time:{seconds:1645961342 nanoseconds:842206000} file:"builtinbackupengine.go" line:148 value:"Hook: , Compress: true"
commerce/0 (zone1-0000000101): time:{seconds:1645961342 nanoseconds:846008000} file:"builtinbackupengine.go" line:158 value:"getting current replication status"
commerce/0 (zone1-0000000101): time:{seconds:1645961342 nanoseconds:887711000} file:"builtinbackupengine.go" line:199 value:"using replication position: 0bd03968-97c0-11ec-aa8e-431ab9537241:1-38"
commerce/0 (zone1-0000000101): time:{seconds:1645961346 nanoseconds:953374000} file:"builtinbackupengine.go" line:294 value:"found 289 files to backup"
commerce/0 (zone1-0000000101): time:{seconds:1645961346 nanoseconds:953466000} file:"builtinbackupengine.go" line:465 value:"Backing up file: _vt/schema_migrations.ibd"
commerce/0 (zone1-0000000101): time:{seconds:1645961346 nanoseconds:953496000} file:"builtinbackupengine.go" line:465 value:"Backing up file: _vt/db.opt"
commerce/0 (zone1-0000000101): time:{seconds:1645961346 nanoseconds:953670000} file:"builtinbackupengine.go" line:465 value:"Backing up file: ibdata1"
[ etc ...]
commerce/0 (zone1-0000000101): time:{seconds:1645961347 nanoseconds:122218000} file:"builtinbackupengine.go" line:465 value:"Backing up file: sys/schema_table_statistics.frm"
commerce/0 (zone1-0000000101): time:{seconds:1645961348 nanoseconds:128686000} file:"builtinbackupengine.go" line:220 value:"resetting mysqld read-only to true"
commerce/0 (zone1-0000000101): time:{seconds:1645961348 nanoseconds:129031000} file:"builtinbackupengine.go" line:237 value:"restarting mysql replication"
➜  local git:(andrew/vtctldserver-backup) ✗  
```

</details>

<details><summary>Backup on primary (fail)</summary>

```
➜  local git:(andrew/vtctldserver-backup) ✗ vtctldclient --server ":15999" Backup zone1-100
E0227 06:30:47.396281    9201 main.go:42] rpc error: code = Unknown desc = TabletManager.Backup on zone1-0000000100 error: type PRIMARY cannot take backup. if you really need to do this, rerun the backup command with -allow_primary: type PRIMARY cannot take backup. if you really need to do this, rerun the backup command with -allow_primary
➜  local git:(andrew/vtctldserver-backup) ✗ 
```

</details>

<details><summary>Backup on primary (success)</summary>

```
➜  local git:(andrew/vtctldserver-backup) ✗ vtctldclient --server ":15999" Backup zone1-100 --allow-primary
commerce/0 (zone1-0000000100): time:{seconds:1645961467 nanoseconds:99775000} file:"builtinbackupengine.go" line:148 value:"Hook: , Compress: true"
commerce/0 (zone1-0000000100): time:{seconds:1645961467 nanoseconds:100657000} file:"builtinbackupengine.go" line:158 value:"getting current replication status"
commerce/0 (zone1-0000000100): time:{seconds:1645961467 nanoseconds:101586000} file:"builtinbackupengine.go" line:179 value:"turning primary read-only before backup"
commerce/0 (zone1-0000000100): time:{seconds:1645961467 nanoseconds:101789000} file:"builtinbackupengine.go" line:199 value:"using replication position: 0bd03968-97c0-11ec-aa8e-431ab9537241:1-38"
commerce/0 (zone1-0000000100): time:{seconds:1645961479 nanoseconds:189094000} file:"builtinbackupengine.go" line:294 value:"found 289 files to backup"
commerce/0 (zone1-0000000100): time:{seconds:1645961479 nanoseconds:189166000} file:"builtinbackupengine.go" line:465 value:"Backing up file: _vt/db.opt"
commerce/0 (zone1-0000000100): time:{seconds:1645961479 nanoseconds:189187000} file:"builtinbackupengine.go" line:465 value:"Backing up file: ibdata1"
commerce/0 (zone1-0000000100): time:{seconds:1645961479 nanoseconds:189314000} file:"builtinbackupengine.go" line:465 value:"Backing up file: ib_logfile0"
[ etc ...]
commerce/0 (zone1-0000000100): time:{seconds:1645961479 nanoseconds:348492000} file:"builtinbackupengine.go" line:465 value:"Backing up file: performance_schema/events_statements_summary_by_program.frm"
commerce/0 (zone1-0000000100): time:{seconds:1645961480 nanoseconds:358154000} file:"builtinbackupengine.go" line:220 value:"resetting mysqld read-only to false"
➜  local git:(andrew/vtctldserver-backup) ✗ 
```

</details>

### **Note to Reviewers**

I'd love some feedback/suggestions on better output; I'm not thrilled about these json blobs (but I _do_ think we should provide an option to use this format for machine/automation processing) for a friendly user-experience, but I also don't have strong opinions.

## Related Issue(s)

- #8231

## Checklist
- [x] Should this PR be backported? **no**
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/cmd/vtctldclient/command/backups.go [hunk 1]
```diff
@@ -28,11 +29,56 @@ import (
...
+	}
+	// GetBackups makes a GetBackups gRPC call to a vtctld.
+	GetBackups = &cobra.Command{
+		Use:  "GetBackups <keyspace/shard>",
+		Args: cobra.ExactArgs(1),
+		RunE: commandGetBackups,
+	}
+)
+
+var backupOptions = struct {
```
Comment (systay) [inline go/cmd/vtctldclient/command/backups.go:49]: nit: this could be added to the var block above

File: go/cmd/vtctldclient/command/backups.go [hunk 2]
```diff
@@ -78,6 +124,10 @@ func commandGetBackups(cmd *cobra.Command, args []string) error {
 }
 
 func init() {
+	Backup.Flags().BoolVar(&backupOptions.AllowPrimary, "allow-primary", false, "Allow the primary of a shard to be used for the backup. WARNING: If using the builtin backup engine, this will shutdown mysqld on the primary and stop writes for the duration of the backup.")
```
Comment (deepthi) [inline go/cmd/vtctldclient/command/backups.go:127]: Right now the flag has an underscore, not dash.

File: go/cmd/vtctldclient/command/backups.go [hunk 3]
```diff
@@ -78,6 +124,10 @@ func commandGetBackups(cmd *cobra.Command, args []string) error {
...
+	Backup.Flags().Uint64Var(&backupOptions.Concurrency, "concurrency", 4, "Specifies the number of compression/checksum jobs to run simultaneously.")
```
Comment (deepthi) [inline go/cmd/vtctldclient/command/backups.go:128]: This syntax is so clean 😃

**Review by deepthi — COMMENTED** [review]
I think I found a typo, the rest lgtm.
Since that is in a flag name, I'm not approving this yet (to avoid merging with the typo by mistake).

File: go/cmd/vtctldclient/command/backups.go (hunk 1 above)
Comment (ajm188) [reply go/cmd/vtctldclient/command/backups.go:49]: I'm trying to match the structure of the other command files, which go like:

```go
var (
    allCommandsGoHere
)

var firstCommandOptions

func firstCommand

func secondCommandWithNoOptions

var thirdCommandOptions

func thirdCommand

func init
```

Comment (ajm188) [reply go/cmd/vtctldclient/command/backups.go:49]: for example: https://github.com/vitessio/vitess/blob/main/go/cmd/vtctldclient/command/cells.go

File: go/cmd/vtctldclient/command/backups.go (hunk 2 above)
Comment (deepthi) [reply go/cmd/vtctldclient/command/backups.go:127]: TIL

**Review by deepthi — APPROVED** [review]

**Pull request merged**